truex-ios-mobile-reference-app/UnlockSoloViewController.m

```diff
@@ -0,0 +1,313 @@
+//
+//  UnlockSoloViewController.m
+//  truex-ios-mobile-reference-app
+//
+//  Created by kyle on 9/27/21.
+//  Copyright © 2021 true[X]. All rights reserved.
+//
+
+#import "UnlockSoloViewController.h"
+#import "WebViewViewController.h"
+#import <TruexAdRenderer/TruexAdRenderer.h>
+
+@interface UnlockSoloViewController ()
+
+@property TruexAdRenderer* activeAdRenderer;
+@property NSMutableDictionary* vastDictionary;
+
+@property (weak, nonatomic) IBOutlet UISwitch *unlocked;
+
+// internal states
+@property NSMutableDictionary* currentPointer;
+
+@end
+
+// internal state for the fake ad manager
+BOOL _vastReady = NO;
+
+
+@implementation UnlockSoloViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    [NSNotificationCenter.defaultCenter addObserver:self
+                                           selector:@selector(pause)
+                                               name:UIApplicationWillResignActiveNotification
+                                             object:nil];
+    [NSNotificationCenter.defaultCenter addObserver:self
+                                           selector:@selector(resume)
+                                               name:UIApplicationDidBecomeActiveNotification
+                                             object:nil];
+}
+
+- (void)viewDidAppear:(BOOL)animated {
+    // Helper function to fetch ad to vastDictionary
+    // This should be pointing to your ad server, where a true[X] ad is booked.
+    [self fetchAd:@"https://qa-get.truex.com/5075c46a8e5a48a206318d4ecfb5cc70101e0bcf/vast/solo?dimension_2=1&stream_position=midroll&stream_id=[stream_id]&network_user_id=[user_id]"];
+}
+
+- (void)viewDidDisappear:(BOOL)animated {
+    [self resetActiveAdRenderer];
+}
+
+- (IBAction)unlockWithTruex:(id)sender {
+    if (self.unlocked.on) {
+        [self alertWithTitle:@"Unlocked" message:@"Already Unlocked" completion:nil];
+        return;
+    }
+    
+    if (!_vastReady || !self.vastDictionary) {
+        [self alertWithTitle:@"Not Ready" message:@"Downloading VAST" completion:nil];
+        return;
+    }
+    
+    // [1] - Look for true[X] in ads
+    // Here we use a fake ad manager, which parse the VAST XML directly into a dictionary.
+    @try {
+        // Just checking the 1st ad here to simplify the flow.
+        NSMutableDictionary* currentAd = self.vastDictionary[@"Ad"][0][@"InLine"][0];
+        BOOL isTruexAd = [self isTruexAd: currentAd];
+        if (isTruexAd) {
+            // [2] - Prepare to enter the engagement
+            [self resetActiveAdRenderer];
+            
+            // Getting the adParameter from current ad, probably will look different in your Ad Framework.
```

Comment:
I'm not sure I understand this comment. Why would the ad payload look different?